clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -1076,6 +1076,16 @@ void CodeGenModule::Release() {
                                 "sign-return-address-with-bkey", 1);
   }
 
+  if (Arch == llvm::Triple::aarch64 || Arch == llvm::Triple::aarch64_be) {
```

Comment:
Stack probing needs to be enabled by module-level attributes or it won't have effect in LLVM generated functions.

Reply:
Module-level attributes tend to lead to issues with LTO.  I mean, for compiler-generated functions, there's no obvious choice, so we just kind of have to pick something.  But for other functions, I think it makes sense to try to respect what the user specified for each input module.  That would suggest we should emit both a module attribute and a function attribute.

Reply:
Done, we now emit both